indra/literature/pubmed_client.py

```diff
@@ -307,37 +307,157 @@ def _find_elem_text(root, xpath_string):
+        the NLM catalog regardless of whether they are found in the XML.
+        Default is 'never' (i.e., never recover from NLM catalog regardless
+        of whether they are found in the XML).
+
+    Returns
+    -------
+    dict
+        A dictionary journal, issue, and ISSN info. The structure is as
+        follows:
+        {
+            "journal_title": str,
+            "journal_abbrev": str,
+            "journal_nlm_id": str,
+            "issn_dict": {
+                "issn": str,
+                "issn_l": str,
+                "type": "print"|"electronic"|"other",
+            },
+            "issue_dict": {
+                "volume": str,
+                "issue": str,
+                "year": int
+            }
+        }
+    """
+    if get_issns_from_nlm not in ['never', 'missing', 'always']:
+        raise ValueError("get_issns_from_nlm must be one of 'never', "
+                         "'missing', or 'always'")
+    # Journal info
+    journal = medline_citation.find('Article/Journal')
+    journal_title = _find_elem_text(journal, 'Title')
+    journal_abbrev = _find_elem_text(journal, 'ISOAbbreviation')
+
+    # Issue info
+    issue_info = _get_issue_info(journal)
+
+    # Get the ISSN from the article record
+    issn_dict = {}
+    issn_element = journal.find("ISSN")
+    if issn_element is not None:
+        issn_type = issn_element.attrib.get("IssnType", "other").lower()
+        issn = issn_element.text
+        issn_dict["issn"] = issn
+        issn_dict["type"] = issn_type
+
+    # Get the linking ISSN from the article record
+    issn_linking = _find_elem_text(medline_citation,
+                                   "MedlineJournalInfo/ISSNLinking")
+    if issn_linking:
+        issn_dict["issn_l"] = issn_linking
+
+    nlm_id = _find_elem_text(medline_citation,
+                             'MedlineJournalInfo/NlmUniqueID')
+
+    # Get ISSN values from the NLM catalog
+    if nlm_id and (
+            get_issns_from_nlm == 'always' or
+            get_issns_from_nlm == 'missing' and not any(issn_dict.values())
+    ):
+        nlm_issn_list = get_issns_for_journal(nlm_id)
+        if nlm_issn_list:
+            issn_dict['alternate_issns'] = nlm_issn_list
+
+    return {
+        "journal_title": journal_title,
+        "journal_abbrev": journal_abbrev,
+        "journal_nlm_id": nlm_id,
+        "issn_dict": issn_dict,
+        "issue_dict": issue_info,
+    }
+
+
+def _get_journal_info(medline_citation, get_issns_from_nlm: bool):
     # Journal info
     journal = medline_citation.find('Article/Journal')
     journal_title = _find_elem_text(journal, 'Title')
     journal_abbrev = _find_elem_text(journal, 'ISOAbbreviation')
 
+    # Issue info
+    issue_info = _get_issue_info(journal)
+
     # Add the ISSN from the article record
-    issn_list = []
+    issn_set = set()
     issn = _find_elem_text(journal, 'ISSN')
     if issn:
-        issn_list.append(issn)
+        issn_set.add(issn)
```

Comment:
do you need to keep track of what the publication type for each ISSN is? maybe this should be a dict?

Reply:
That ISSN type is available if using `get_issn_info` further up in the file.

The function where this line is from (`_get_journal_info`) is nested in other functions where there is an expectation of `issn_list` being present as a list of ids, so changing code in this function would potentially include updating downstream usage as well so I'd argue it's outside the scope of this PR.